### DISCUSSION TITLE
Does it support OpenTelemetry Protocol (OTLP)?

### DISCUSSION THREAD

**Discussion by ejsmith** [openobserve/openobserve#935] [Q&A] [answered]
From what I can tell, it appears there is some OpenTelemetry support, but it does not support the OTLP. Is that correct?

**Comment by hengfeiyang** [comment]
[Logs or Metrics?](https://openobserve.ai/docs/ingestion/metrics/otel_collector/)

**Comment by ejsmith** [comment]
OTLP is for logs, metrics and traces and it's a single GRPC endpoint. I noticed that there is OpenTelemetry support for metrics and tracing, but they are different endpoints and metrics is using otel collector with a prometheus exporter. Ideally, openobserve would support OTLP fully with a single endpoint and I can send my metrics directly to it from my app or I can use an OTel collector instance in between.

**Comment by prabhatsharma** [comment, marked as answer]
OTLP support is currently available but only for traces. Logs and metrics support for OTLP will be available soon.

**Reply by rgidda** [reply]
How soon OTLP support is available ?

**Reply by prabhatsharma** [reply]
a month to a couple of months. Too many high priority items for now. You can track it at https://github.com/openobserve/openobserve/issues/981

**Reply by kirinnee** [reply]
Hello, I just want to check if OTLP is available already? #981 seems to be closed, but I still cannot find any documentation of this feature

**Reply by prabhatsharma** [reply]
Updated docs - https://openobserve.ai/docs/ingestion/logs/otlp/ . You should be able to see more details, including credentials  once you login to your cloud/self hosted instance.
